### PR TITLE
AST-45352 - Add support for Checkmarx Azure AI

### PR DESCRIPTION
**Pull request by avisab-cx**
By submitting a PR to this repository, you agree to the terms within the [Checkmarx Code of Conduct](../docs/code_of_conduct.md). Please see the [contributing guidelines](../docs/contributing.md) for how to create and submit a high-quality PR for this repo.

### Description

> Describe the purpose of this PR along with any background information and the impacts of the proposed change.

### References

> Include supporting link to GitHub Issue/PR number

### Testing

> Describe how this change was tested. Be specific about anything not tested and reasons why. If this solution has unit and/or integration testing, tests should be added for new functionality and existing tests should complete without errors.
>
> Please include any manual steps for testing end-to-end or functionality not covered by unit/integration tests.

### Checklist

- [ ] I have added documentation for new/changed functionality in this PR (if applicable).
- [ ] I have updated the CLI help for new/changed functionality in this PR (if applicable).
- [ ] All active GitHub checks for tests, formatting, and security are passing
- [ ] The correct base branch is being used

### CODE REVIEW

File: internal/commands/chat-kics.go [hunk 1]
```diff
@@ -104,9 +130,39 @@ func runChatKics(chatKicsWrapper wrappers.ChatWrapper) func(cmd *cobra.Command,
...
+		tenantID, _ := wrappers.ExtractFromTokenClaims(customerToken, tenantIDClaimKey)
+		// remove from tenant id all the string before ::
+		if strings.Contains(tenantID, "::") {
+			tenantID = tenantID[strings.LastIndex(tenantID, "::")+2:]
+		}
+
+		requestID := statefulWrapper.GenerateId().String()
+
+		var response []message.Message
+		if azureAiEnabled || checkmarxAiEnabled {
```
[the remark this answers is not in the document]
Comment (avisab-cx) [reply internal/commands/chat-kics.go:142]: no. it's not possible. it's a radio button in the configuration UI. if one is true, the other must be false.

File: internal/commands/chat-sast.go [hunk 2]
```diff
@@ -101,9 +110,37 @@ func runChatSast(chatWrapper wrappers.ChatWrapper, tenantWrapper wrappers.Tenant
...
+
+		var response []message.Message
+		if azureAiEnabled || checkmarxAiEnabled {
+			metadata := message.MetaData{
+				TenantID:  tenantID,
+				RequestID: requestID,
+				UserAgent: params.DefaultAgent,
+				Feature:   guidedRemediationFeatureNameSast,
+			}
+			if azureAiEnabled {
```
[the remark this answers is not in the document]
Comment (avisab-cx) [reply internal/commands/chat-sast.go:127]: same answer :)

File: internal/commands/chat-sast.go [hunk 3]
```diff
@@ -38,75 +41,51 @@ func ChatSastSubCommand(chatWrapper wrappers.ChatWrapper, tenantWrapper wrappers
...
-				Role:    role.User,
-				Content: userPrompt,
-			})
-		} else {
-			newMessages = append(newMessages, message.Message{
-				Role:    role.User,
-				Content: userInput,
-			})
+		newConversation, userInput, id, err4, done3 := getSastConversationDetails(cmd, chatConversationID, statefulWrapper)
+		if done3 {
```
Comment (AlvoBen) [inline internal/commands/chat-sast.go:75]: please change to more indicative name: sastConversationErr, sastConversationSuccess and etc

File: internal/commands/chat-sast.go [hunk 4]
```diff
@@ -38,75 +41,51 @@ func ChatSastSubCommand(chatWrapper wrappers.ChatWrapper, tenantWrapper wrappers
...
+			return err4
 		}
 
-		response, err := chatWrapper.Call(statefulWrapper, id, newMessages)
-		if err != nil {
-			return outputError(cmd, id, err)
+		newMessages, err2, done := buildSastMessages(cmd, newConversation, scanResultsFile, sastResultID, sourceDir, id, userInput)
```
Comment (AlvoBen) [inline internal/commands/chat-sast.go:79]: also here

File: internal/commands/chat-sast.go [hunk 5]
```diff
@@ -38,75 +41,51 @@ func ChatSastSubCommand(chatWrapper wrappers.ChatWrapper, tenantWrapper wrappers
...
+		if done {
+			return err2
 		}
 
-		responseContent := getMessageContents(response)
+		responseContent, done2, err3 := sendRequest(cmd, statefulWrapper, azureAiEnabled, checkmarxAiEnabled, tenantID, chatWrapper,
```
Comment (AlvoBen) [inline internal/commands/chat-sast.go:84]: and here

File: internal/commands/chat-sast.go [hunk 6]
```diff
@@ -117,23 +96,152 @@ func runChatSast(chatWrapper wrappers.ChatWrapper, tenantWrapper wrappers.Tenant
...
+	customerToken, _ = wrappers.GetAccessToken()
+
+	if azureAiEnabled {
+		aiProxyAzureAIRoute := viper.GetString(params.AiProxyAzureAiRouteKey)
+		aiProxyEndPoint, _ := wrappers.GetURL(aiProxyAzureAIRoute, customerToken)
+		model, _ := GetAzureAiModel(tenantConfigurationResponses)
+		statefulWrapper, _ = wrapper.NewStatefulWrapperNew(conn, aiProxyEndPoint, customerToken, model, dropLen, 0)
+	} else if checkmarxAiEnabled {
+		aiProxyCheckmarxAIRoute := viper.GetString(params.AiProxyCheckmarxAiRouteKey)
+		aiProxyEndPoint, _ := wrappers.GetURL(aiProxyCheckmarxAIRoute, customerToken)
```
Comment (AlvoBen) [inline internal/commands/chat-sast.go:162]: look at main.go, you should had to chatWrapper constructor this paths as params and use it from the chatWrapper object like feilds

File: internal/params/envs.go [hunk 7]
```diff
@@ -61,5 +61,8 @@ const (
 	AccessManagementPathEnv             = "CX_ACCESS_MANAGEMENT_PATH"
 	ByorPathEnv                         = "CX_BYOR_PATH"
 	IgnoreProxyEnv                      = "CX_IGNORE_PROXY"
+	VorpalPortEnv                       = "CX_VORPAL_PORT"
```
Comment (AlvoBen) [inline internal/params/envs.go:64]: delete - old logic, probably merge conflicts

File: internal/params/keys.go [hunk 8]
```diff
@@ -61,5 +61,8 @@ var (
 	PolicyEvaluationPathKey             = strings.ToLower(PolicyEvaluationPathEnv)
 	AccessManagementPathKey             = strings.ToLower(AccessManagementPathEnv)
 	ByorPathKey                         = strings.ToLower(ByorPathEnv)
+	VorpalPortKey                       = strings.ToLower(VorpalPortEnv)
```
Comment (AlvoBen) [inline internal/params/keys.go:64]: also here

File: internal/wrappers/client.go [hunk 9]
```diff
@@ -35,6 +35,7 @@ const (
 	tryPrintOffset          = 2
 	retryLimitPrintOffset   = 1
 	MissingURI              = "When using client-id and client-secret please provide base-uri or base-auth-uri"
+	BaseUriisEmpty          = "Base URI is empty"
```
Comment (AlvoBen) [inline internal/wrappers/client.go:38]: BaseUriIsEmpty*

File: internal/params/binds.go [hunk 10]
```diff
@@ -62,5 +62,8 @@ var EnvVarsBinds = []struct {
 	{PolicyEvaluationPathKey, PolicyEvaluationPathEnv, "api/policy_management_service_uri/evaluation"},
 	{AccessManagementPathKey, AccessManagementPathEnv, "api/access-management"},
 	{ByorPathKey, ByorPathEnv, "api/byor"},
+	{VorpalPortKey, VorpalPortEnv, ""},
```
Comment (AlvoBen) [inline internal/params/binds.go:65]: delete also

**Review by AlvoBen — DISMISSED** [review]

**Review by AlvoBen — APPROVED** [review]

**Pull request merged**